src/backend/joanie/core/api/client.py

```diff
@@ -75,10 +75,52 @@ def get_queryset(self):
             if self.request.auth
             else self.request.user.username
         )
+        queryset = models.CourseProductRelation.objects
+        organization_id = self.kwargs.get("organization_id", None)
+        if organization_id:
+            queryset = queryset.filter(
+                organizations__id=organization_id,
+                organizations__accesses__user__username=username,
+            )
+        else:
+            queryset = queryset.filter(course__accesses__user__username=username)
+        return queryset.select_related("course", "product").prefetch_related(
+            "course__course_runs",
+            "course__organizations",
+            "course__products",
+            "organizations",
+            "product__target_courses",
+        )
```

Comment:
Are we sure `prefetch_related` is not counterproductive ?

Reply:
In my tests it reduces the amount of queries, sometime significantly (from 50ish to 30ish). Maybe it takes more time overall, I'll check that